### ISSUE TITLE
cache integrator JIT

### ISSUE THREAD

**Issue by ghorn** [casadi/casadi#1955]
It's annoying to do <up> <enter> to re-run a simulation, and then wait for integrator JIT, especially with high compiler optimization level. A caching mechanism would be nice.

Or could it be implemented on the user side with external function?

**Comment by jaeandersson** [comment]
Idk... JIT isn't really too suitable for rapid prototyping. How long does compilation take for you?

**Comment by jgillis** [comment]
This can be done wiyh shell jit compiler. You can put a cacher as compiler

**Comment by jgillis** [comment]
Ill send code

**Comment by jgillis** [comment]
Greg has fixed this 0eab93e951b709c091c77772078757dcc0c3d96b


Test code:
```python
from casadi import *


x = SX.sym("x",8,8)
d = det(x)
import time

t0 = time.time()
f = Function('f',[x],[d],{"jit":True,"compiler":"shell","jit_options": {"compiler": "gcc", "compiler_flags":["-O3"]}})
print time.time()-t0
t0 = time.time()
f = Function('f',[x],[d],{"jit":True,"compiler":"shell","jit_options": {"compiler": "ccache gcc", "compiler_flags":["-O3"]}})
print time.time()-t0
```
